pkg/utils/predicates_test.go

```diff
@@ -1099,12 +1101,61 @@ func TestCheckPodsWithAntiAffinityExist(t *testing.T) {
 			}), "foo", "bar"),
 			podsInNamespace: map[string][]*v1.Pod{},
 			nodeMap:         map[string]*v1.Node{},
+			nodeFit:         false,
 			expMatch:        false,
 		},
+		{
+			name: "found pod matching pod anti-affinity when node fitting",
+			pod:  test.PodWithPodAntiAffinity(test.BuildTestPod("p1", 1000, 1000, "node", nil), "foo", "bar"),
+			podsInNamespace: map[string][]*v1.Pod{
```

Comment:
Unless a pod is scheduled to a node that no longer has TopologyKey label key, pod anti-affinity needs presence of at least two pods to violate the rule. The same holds for other newly introduced test cases.